File: N2/ProyectoFinal_Alba_Torres/JAVA/demojpa/src/main/java/service/ProductoService.java
package service;
import java.util.List;

import javax.inject.Inject;
import javax.ws.rs.Consumes;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import DAO.ProductoDao;
import ec.edu.ups.pweb.demojpa.Producto;

@Path("productos")
public class ProductoService {

	@Inject
	ProductoDao daoProducto;
	
	@POST
	@Path("newPro")
	@Produces(MediaType.APPLICATION_JSON)
	@Consumes("application/json")
	public String guardarProducto(Producto p) {
		try {
			daoProducto.insert(p);
		}catch(Exception e){
			return "{\"resultado\": \"ERROR\"}";
		}
		return "{\"resultado\": \"OK\"}";
	}

	@POST
	@Path("findPro")
	@Produces(MediaType.APPLICATION_JSON)
	@Consumes("application/json")
	public Producto buscarProducto(int codigo) {
		return daoProducto.read(codigo);		
	}

	@POST
	@Path("updatePro")
	@Produces(MediaType.APPLICATION_JSON)
	@Consumes("application/json")
	public String actualizarPro(Producto p) {
		try {
			daoProducto.update(p);
		}catch(Exception e){
			return "{\"resultado\": \"ERROR\"}";
		}
		return "{\"resultado\": \"OK\"}";
		
		
	}
	@POST
	@Path("deletePro")
	@Produces(MediaType.APPLICATION_JSON)
	@Consumes("application/json")
	public String eliminarPro(int codigo) {
		try {
			System.out.println(codigo);
			daoProducto.delete(codigo);
		}catch(Exception e){
			return "{\"resultado\": \"ERROR\"}";
		}
		return "{\"resultado\": \"OK\"}";
		
		
	}
	
	@GET
	@Path("listPro")
	@Produces(MediaType.APPLICATION_JSON)
	public List<Producto> getListPer(){
		return daoProducto.getList();
	}
	//Content-type=application/json
}

File: N2/ProyectoFinal_Alba_Torres/Angular/src/app/services/producto.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Producto } from '../domain/producto';

@Injectable({
  providedIn: 'root'
})
export class ProductoService {

  
  constructor(private http: HttpClient) { }

  guardar(p: Producto): Observable<any[]>{
    return this.http.post<any>("http://localhost:8080/demojpa/rs/productos/newPro", p)
  }
  editar(p: Producto): Observable<any[]>{
    return this.http.post<any>("http://localhost:8080/demojpa/rs/productos/updatePro", p)
  }
  listar(): Observable<any[]>{
    return this.http.get<any>("http://localhost:8080/demojpa/rs/productos/listPro")
  }
  delete(p: number): Observable<any[]>{
    
    return this.http.post<any>("http://localhost:8080/demojpa/rs/productos/deletePro", p)
  }
  buscar(codigo: number): Observable<any[]>{ 
    return this.http.post<any>("http://localhost:8080/demojpa/rs/productos/findPro", codigo)
  }

}
